lib/csv_export.rb

```diff
@@ -0,0 +1,70 @@
+module CsvExport
+  require 'csv'
+  require 'zip'
+
+  # generates a csv from the given model data
+  def generate_csv(data)
+    objects = data.first
+    columns = data.size == 1 ? objects.first.attributes.keys : data[1]
+
+    # we never need to export ids
+    columns.delete('id')
+
+    CSV.generate(headers: true) do |csv|
+      csv << columns
+
+      objects.each do |o|
+        csv << columns.map do |attr|
+          s = o.send(attr)
+          if s.is_a? ActiveRecord::Base
+            s.name
+          else s
+          end
+        end
+      end
+    end
+  end
+
+  # generates a zip file containing multiple CSVs
+  def generate_zip(data)
+    # create the CSVs
+    csvs = data.map { |model| generate_csv(model) }
+
+    Zip::OutputStream.write_buffer do |stream|
+      csvs.each_with_index do |csv, i|
+        model_name = data[i].first.first.class.name
```

Comment:
We could just include this in the `generate_csv` model for consistency across different use cases.